### ISSUE TITLE
SFTP - Use sudo on Login

### ISSUE THREAD

**Issue by Chevrah** [zhanghai/MaterialFiles#918]
With WinSCP there's an option to to start a session with sudo, can I do it with MaterialFiles?

https://winscp.net/eng/docs/faq_su#sudo

**Comment by zhanghai** [comment]
That page says with SFTP you can start the sftp-server as root, so that's something you need to do on your server and can't be done in Material Files.

**Comment by PaperStrike** [comment]
@zhanghai This is actually possible and I believe the page isn't saying something server-side. Well I have never used WinSCP, but Xftp also provides a very similar client-side feature called "custom sftp-server" and I confirm that it works.

After some research on how sftp works, I locally modified [`app/src/main/java/me/zhanghai/android/files/provider/sftp/client/Client.kt`](https://github.com/zhanghai/MaterialFiles/blob/5620678c309842135da46233a368b4c7b01c660c/app/src/main/java/me/zhanghai/android/files/provider/sftp/client/Client.kt#L240) and got a hard-coded working one on my machine:
![image](https://github.com/zhanghai/MaterialFiles/assets/22674396/28cfc4a5-8894-4a49-b8c2-de74b52b1702)

Long story short, the default sftp connection basicly starts with `ssh xxx@xxx -s sftp`, and it is equivalent to `ssh xxx@xxx "/path/to/sftp-server"` (assuming the server's sshd_config contains `Subsystem sftp  /path/to/sftp-server`). So by changing the subsystem call to a custom command call, we can customize the binary we use and/or get privileged access.
